packages/synthetix-main/test/contracts/modules/SNXModule.test.js

```diff
@@ -48,30 +50,56 @@ describe('SNXTokenModule', function () {
         assert.equal(address, snxTokenAddress);
       });
 
+      it('reads the SNX parameters', async () => {
+        assert.equal(await SNX.name(), 'Synthetix Network Token');
+        assert.equal(await SNX.symbol(), 'snx');
+        assert.equal(await SNX.decimals(), 18);
+      });
+
       describe('When attempting to create the SNX twice', () => {
         it('reverts', async () => {
           await assertRevert(SNXTokenModule.connect(owner).createSNX(), 'SNXAlreadyCreated()');
         });
       });
 
       describe('When attempting to upgrade to a new implementation', () => {
-        let AnotherSNXToken;
+        let AnotherSNXToken, NewSNX;
 
         before('Deploy new implementation', async () => {
-          const factory = await ethers.getContractFactory('SNXToken');
+          const factory = await ethers.getContractFactory('SNXTokenMock');
           AnotherSNXToken = await factory.deploy();
         });
 
         before('Upgrade to new implementation', async () => {
           const tx = await SNXTokenModule.connect(owner).upgradeSNXImplementation(
             AnotherSNXToken.address
           );
+
           await tx.wait();
         });
 
         it('is upgraded', async () => {
-          const SNXToken = await ethers.getContractAt('SNXToken', snxTokenAddress);
-          assert.equal(AnotherSNXToken.address, await SNXToken.getImplementation());
+          NewSNX = await ethers.getContractAt('SNXTokenMock', snxTokenAddress);
+          assert.equal(AnotherSNXToken.address, await NewSNX.getImplementation());
+        });
+
+        it('reads the upgraded SNX parameters', async () => {
+          assert.equal(await NewSNX.name(), 'Synthetix Network Token');
+          assert.equal(await NewSNX.symbol(), 'snx');
+          assert.equal(await NewSNX.decimals(), 18);
+        });
+
+        describe('New SNX can mint', () => {
```

Comment:
I'm having second thoughts about this test. I'd like to have it as a integration test / proof of concept, but every part is already covered.

@i-stam / @ajsantander  your thoughts?

Reply:
Yeah, I suppose that this, like other tests, provide "peace of mind". It's not about coverage IMO, it's about proving a point. So I'm for keeping it here 👍